[Chow Jia Ying] Duke Increments

C-likethis123 opened the pull request.

j-lum committed:
Add support for Gradle workflow

j-lum committed:
Add sample checkstyle configuration

Add toolVersion block in to Gradle code sample to prevent errors.

C-likethis123 committed:
Greet the user on startup, echo user messages and display messages on exit

C-likethis123 committed:
Indent all messages displayed by program

C-likethis123 committed:
Add user inputs as tasks and display list of tasks

C-likethis123 committed:
Set Tasks as done and output tasks with their statuses

C-likethis123 committed:
Mark Tasks as Done,Tasks shown with status icons and accompanying messages
* Status icons are ticks for completed Tasks and crosses for uncompleted ones
* Accompanying messages are 'Here are the tasks in your list:' and 'Nice! I've marked this task as done:'

C-likethis123 committed:
Represent ToDo tasks with a ToDo class

C-likethis123 committed:
Represent Event tasks with an Event class

C-likethis123 committed:
Represent Deadline tasks with a Deadline class

C-likethis123 committed:
Supports the addition of ToDo, Event and Deadline tasks
* [T],[D],[E] icons are displayed with the task description
* Event and Deadline tasks can be tagged with a date

C-likethis123 committed:
Fix indentation errors

C-likethis123 committed:
Add semi-automated testing of Duke
* runbat.sh compiles and compares the expected output to actual output
* EXPECTED.txt contains expected output
* ACTUAL.txt contains actual output of the program

C-likethis123 committed:
Creates the following DukeExceptions that will be raised when the program encounters incorrect user inputs:
* EmptyToDoDescriptionException
* UnknownCommandException

C-likethis123 committed:
Handle Errors due to incorrect user inputs

C-likethis123 committed:
Delete tasks from the taskList

C-likethis123 committed:
Fix error where delete function deletes the wrong task from the taskList

C-likethis123 committed:
Added a text file to store list of tasks
* Load tasks in text file upon starting Duke program
* Update text file upon change of task list

C-likethis123 committed:
Recognise dates and times with DateTime class

C-likethis123 committed:
Deadlines and Events in Duke recognises dates and times

C-likethis123 committed:
Implement saving tasks to hard disk, loading tasks upon program start

C-likethis123 committed:
Merge branch 'branch-Level-7'

C-likethis123 committed:
Emptied ACTUAL.TXT and EXPECTED.txt

C-likethis123 committed:
Merge branch 'branch-Level-8'

C-likethis123 committed:
Create Storage class in charge of loading and saving tasks

C-likethis123 committed:
Fixed formatting for toString method for Event and Deadline classes

C-likethis123 committed:
Add ExitCommand class to represent exit commands

C-likethis123 committed:
Fix error when loading from storage

C-likethis123 committed:
Fix error when loading from storage

C-likethis123 committed:
Fix error when loading from storage

C-likethis123 committed:
Merge branch 'master' of https://github.com/C-likethis123/duke
Apply changes from remote repo

C-likethis123 committed:
Extracted code into 5 classes:
* Commands - represent user commands
* Ui - deals with interaction with user
* Storage - deals with writing to the file and saving tasks to the file
* Parser - deals with making sense of the user command
* TaskList - contains the tasks list

C-likethis123 committed:
Added DateTimeTest class for testing

C-likethis123 committed:
Merge branch 'gradle'

C-likethis123 committed:
Fixed bug where task cannot be deleted

C-likethis123 committed:
Added JavaDoc comments

C-likethis123 committed:
Tweaked code to comply to coding standard

C-likethis123 committed:
Add Search functionality for tasks

C-likethis123 committed:
Added Javadoc comments for DoneCommand and Storage class

C-likethis123 committed:
Made minor edits to Javadoc comments

C-likethis123 committed:
Fixed according to coding standard

C-likethis123 committed:
Fixed conflicts in Ui class for merge

C-likethis123 committed:
Fixed minor error in comment

C-likethis123 committed:
Include JavaDoc comments to classes added in Level-9

C-likethis123 committed:
Enable JUnit testing with Gradle

C-likethis123 committed:
Create GUI using JavaFX

* Launcher class is the entry point of the application
* Duke class has a start() method, which sets and renders a Stage.

C-likethis123 committed:
Add more components to JavaFX GUI

Components added:
* ScrollPane for scrolling
* TextField for user to type inputs
* Send button to send inputs (not working yet)

walnads commented on src/main/java/DateTime.java:
```diff
@@ -0,0 +1,99 @@
+/**
+ * DateTime represents dates specified in Deadlines and Event Tasks.
+ */
+public class DateTime {
```
Just curious why you didn't use the DateTime class to implement this instead!

wpuffs replied:
Hi, just curious why you didn't use the DateTime class to implement this instead!

C-likethis123 committed:
Add Dialog Boxes to GUI and handle user inputs

A DialogBox is a custom control consisting of an ImageView control and Label control. With DialogBoxes, user inputs and Duke replies are accompanied by avatars.

User inputs are also handled via the handleUserInput() method, where Duke sends a message to the user whenever Duke receives user inputs.

C-likethis123 committed:
Differentiate User and Duke messages

* User messages have an avatar on the right, while Duke messages have
an avatar on the left.
* DialogBox class have separate DialogBoxes for users and Duke

wpuffs commented on src/main/java/ToDo.java:
```diff
@@ -0,0 +1,9 @@
+public class ToDo extends Task {
+    public ToDo(String description) {
+        super(description);
+        this.typeOfTask = "[T]";
```
Not really a suggestion again, just curious why there's a need for a typeOfTask variable when you already have the different types of tasks as individual classes inheriting the Task class!

wpuffs reviewed (commented):
Hey Jia Ying,

Overall a job well done! I liked how there was extensive OOP usage and your Javadocs was clear in explaining what each class and method does. Just a few comments, 
- I think you can consider adding files like graddle-wrapper.jar, graddle-wrapper.properties, gradlew, etc. to your .gitignore
-  You have a few Command class files, you could consider packaging them into duke.command so that it looks slightly neater! 😄

C-likethis123 committed:
Refine look of DialogBox in GUI

* Avatars are now circles instead of squares
* Added padding between text and avatar
* Added padding between DialogBoxes

C-likethis123 committed:
Configure look of GUI with FXML

* The main window look is configured by MainWindow.fxml
* The dialog box look is configured by DialogBox.fxml

C-likethis123 committed:
Add controller for MainWindow.fxml

Purpose of adding MainWindow.java is to separate code
handling Duke's logic and UI.

C-likethis123 committed:
Fix bug in handleUserInput() method

Previously the first argument passed into getUserDialog and getDukeDialog methods
were Strings, but the method signatures takes in Labels instead.

The change is to create Label objects out of these Strings and pass these Label objects
into the getUserDialog and getDukeDialog methods.

C-likethis123 committed:
Modify access modifier of getResponse method

Previously, the handleUserInput method of MainWindow.java could not use the
getResponse method because it was set to private.

The change is to set it to protected. Future changes to the getResponse method would be to change the output of the method.

C-likethis123 committed:
Bridge application logic and UI with Main class

Duke.java handles the application logic, while GUI is configured in
MainWindow.java. This class is created, with the start() method being the
entry point of the application.

C-likethis123 committed:
Set Main class as class to launch

Previously, the Duke class was the class to launch. Now the Main class acts as
entry point to the program instead of Duke.

C-likethis123 committed:
Modify handleUserInput method

C-likethis123 committed:
Add controller for DialogBox.fxml

C-likethis123 committed:
Remove GUI configurations in Duke class

C-likethis123 committed:
Add padding between ImageView and Label components

C-likethis123 committed:
Display welcome messsage when application launches

C-likethis123 committed:
Change getResponse method and package classes

C-likethis123 committed:
Import DateTime class to the DateTimeTest class

The change is needed because DateTime is now packaged.

C-likethis123 committed:
Change the controller of MainWindow.fxml

MainWindow.java is packaged into duke.controllers.

C-likethis123 committed:
Display sad face properly

jxofficial commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,37 @@
+
+    @Override
+    public String toString() {
+        String statusIcon = isDone ? "[\u2713]" : "[\u2718]"; // \u2713 represents a tick, \u2718 represents a cross
```
Might have exceeded character count for this line

jxofficial reviewed (commented):
Hi Jia Ying, 

I was very impressed with your code; the code in each class corresponds directly with my expectation of what that class should do. I especially like how your` UI `class handles the displaying of the errors passed, instead of leaving it to the `Exception` classes to print out the error. Also, your use of the `DateTime` class is really quite inspired, with many useful methods which cleans up the input dates well.

I just have one small concern. Your `showError` method in the `UI` class seems to be unused. Correct me if I'm wrong! 

Otherwise, I feel that your code is ready for production use. 

Looking forward to your reply.

Jun Xian

C-likethis123 committed:
Display long messages fully

Previously long messages will be truncated with a '...' at the end of the message.

Messages are displayed in a DialogBox control, which previously consists of a Label
object and an Image. I changed the Label to a Text object.

C-likethis123 committed:
Resize controls when window is maximised

Previously when the window is maximised, the controls stay the same size.

What was changed is to set the AnchorPane.bottomAnchor, AnchorPane.leftAnchor,
AnchorPane.rightAnchor, AnchorPane.topAnchor properties to the controls.

C-likethis123 committed:
Show error when user asks for non-existant task

Previously when a user asks to perform an action on a task that
does not exist (eg 'delete 100'), the user input box does not clear.

Now the application responds that the task does not exist.

C-likethis123 committed:
Modify representation of tabs

Previously tabs were represented by a TABS constant.

It was replaced by the '\t' special character

C-likethis123 committed:
Add assertions (#4)

C-likethis123 committed:
Refactored code (#5)

* Remove 'null' displayed at start of messages

* Refactored Parser.java

C-likethis123 committed:
Update build.gradle

C-likethis123 committed:
Update build.gradle

C-likethis123 committed:
Merge branch 'master' of https://github.com/C-likethis123/duke

C-likethis123 committed:
Set up Travis for CI

* Set up Travis for CI

* Correct minor spacing error

* Update gradlew to allow permissions

* Update checkstyle settings

C-likethis123 committed:
Add user command to update tasks

C-likethis123 committed:
Add Javadoc comments on UpdateCommand.java

C-likethis123 committed:
Fix bug where empty ToDos are not detected

C-likethis123 committed:
Specify file path relative to directory

C-likethis123 committed:
Add Javadoc comments

C-likethis123 committed:
Improve code quality

C-likethis123 committed:
Add WrongEventFormatException to throw errors for wrong Update Commands

C-likethis123 committed:
Modify prompt text in text box

C-likethis123 committed:
Add representative screenshot

C-likethis123 committed:
Fix Travis build fail

C-likethis123 committed:
Create initial user guide

C-likethis123 committed:
Update how to set up Duke

C-likethis123 committed:
Update build.gradle

C-likethis123 committed:
Add a link to download jar file

C-likethis123 committed:
Set theme jekyll-theme-cayman

C-likethis123 committed:
Add screenshot showing list of tasks

C-likethis123 committed:
Merge branch 'master' of https://github.com/C-likethis123/duke

C-likethis123 committed:
Add screenshot to user guide

C-likethis123 committed:
Fix look of screenshot in user guide

C-likethis123 committed:
Add image showing Duke start up

C-likethis123 committed:
Add screenshot showing Duke starting up

C-likethis123 committed:
Add Table of Contents to user guide

C-likethis123 committed:
Remove a tags from titles

Previously the titles of the guides had brackets like <a name=quick-start>.

C-likethis123 committed:
Make jar file cross platform

C-likethis123 committed:
Add method to check if data file exists

C-likethis123 committed:
Update user guide
